### PR TITLE
Alerts: Send an email if a received data point fails to pass the defined rules #26 

### PR DESCRIPTION
**Pull request by cgl**
Sends an email if a received data point fails to pass the defined rules.

New Alert Rules can be defined as below:

Alert_Rule.objects.create(site = site, check_field="soc", value=30, operator="gt")
Alert_Rule.objects.create(site = site, check_field="soc", value=35.5, operator="eq")
Alert_Rule.objects.create(site = site, check_field="battery_voltage", value=25, operator="lt")

Rules are based on sites. So you need to define a rule for each site if you want to check same configurations in several sites.

For now system sends an alert email to the address seshdash@gmail.com
A new issue on "site user integration to Alerts" should be opened.

tests.py should be tested

### CODE REVIEW

File: seshdash/utils/alert.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from seshdash.models import Sesh_Site,Site_Weather_Data,BoM_Data_Point, Alert_Rule, Sesh_Alert
+from seshdash.utils.send_mail import send_mail
+from datetime import datetime
+
+def alert_check(data_point):
+    ops = {'lt': lambda x,y: x<y,
+           'gt': lambda x,y: x>y,
+           'eq' : lambda x,y: x==y,
+    }
+    rules = Alert_Rule.objects.filter(site = data_point.site)
```
Comment (jlark) [inline seshdash/utils/alert.py]: A few comments in here would be helpful

File: seshdash/tests.py [hunk 2]
```diff
@@ -1,3 +1,24 @@
 from django.test import TestCase
+from seshdash.models import Sesh_Alert, Alert_Rule, Sesh_Site, BoM_Data_Point as Data_Point
+from seshdash.utils import alert
+from datetime import datetime
 
-# Create your tests here.
+class AlertTestCase(TestCase):
+    site = Sesh_Site.objects.create(site_name=u"Test site", comission_date=datetime(2015, 12, 11, 22, 0),
```
Comment (jlark) [inline seshdash/tests.py]: Thanks for these tests!

A few comments explaining the tests would be helpful. I know it's pretty trivial right now but it won't be forever.

File: seshdash/utils/send_mail.py [hunk 3]
```diff
@@ -3,11 +3,12 @@
...
-def send_mail(subject,to_email):
+def send_mail(subject,to_email,content):
     try:
         plaintext = get_template('mail/site_mail.txt')
         htmly     = get_template('mail/site_mail.html')
-        d = Context({ 'username' : "User2"
+        d = Context({ 'username' : "User2",
+                      'content'  : content,
                       })
         from_email = "seshdash@gmail.com"
```
Comment (jlark) [inline seshdash/utils/send_mail.py]: Aren't these in the configuration file? from_email

File: seshdash/models.py [hunk 4]
```diff
@@ -45,18 +45,27 @@ class Sesh_User(models.Model):
...
+                                      choices=OPERATOR_CHOICES,
+                                      default="lt")
+    send_mail = models.BooleanField(default=True)
+    #TODO a slug field with the field operator and value info can be added
     #TODO this is vastly incomplete!! fields need to be mapable and chooices need to exist
+    def __str__(self):
+        return "site:%s rule:[%s '%s' %s]" %(self.site.site_name,self.check_field,self.operator,self.value)
 
 #TODO Add alert Object to save alerts
 class Sesh_Alert(models.Model):
```
Comment (jlark) [inline seshdash/models.py]: Is this empty for a reason? is it a TODO?

**Comment by jlark** [comment]
Thanks Cagil, Please see my comments.

File: seshdash/utils/send_mail.py (hunk 3 above)
Comment (cgl) [inline seshdash/utils/send_mail.py]: No there is no from_email in the settings. There is EMAIL_HOST_USER : seshdash@gmail.com

File: seshdash/models.py (hunk 4 above)
Comment (cgl) [inline seshdash/models.py]: Which line?

File: seshdash/utils/alert.py (hunk 1 above)
Comment (cgl) [inline seshdash/utils/alert.py]: Done

File: seshdash/tests.py (hunk 2 above)
Comment (cgl) [inline seshdash/tests.py]: done

**Comment by cgl** [comment]
Done some fixes, added some comments.

File: seshdash/models.py (hunk 4 above)
Comment (jlark) [inline seshdash/models.py]: Sesh_Alert class seems to be not implemented.

File: seshdash/utils/send_mail.py (hunk 3 above)
Comment (jlark) [inline seshdash/utils/send_mail.py]: Maybe we can add it to the local_settings.ini file then pull it in as a Constntin the settings.
Once it's a constant in the settings you can access it from the function.  I used to do something similar with the enphase api key.

**Comment by jlark** [comment]
More comments inline

File: seshdash/models.py (hunk 4 above)
Comment (jlark) [inline seshdash/models.py]: Line 71. Class Sesh_Alert seems to be not implemented.

File: seshdash/utils/send_mail.py (hunk 3 above)
Comment (cgl) [inline seshdash/utils/send_mail.py]: DONE

File: seshdash/models.py (hunk 4 above)
Comment (cgl) [inline seshdash/models.py]: You have implemented Sesh_Alert model before. I just used it as is. 

```python
class Sesh_Alert(models.Model):
    site = models.ForeignKey(Sesh_Site)
    alert = models.ForeignKey(Alert_Rule)
    date = models.DateTimeField()
    isSilence = models.BooleanField()
    alertSent = models.BooleanField()
```

Currently when a rule trigger an alert a Sesh_Alert object is created in alert.py Line 27
```python
            ...
            alert_obj = Sesh_Alert.objects.create(site = data_point.site, alert=rule, date=timezone.now(),
                                                  isSilence=True,alertSent=rule.send_mail)
            if rule.send_mail:
                mail_sent = alert(data_point,content,recipients)
                alert_obj.alertSent = mail_sent
                print("Sent mail for %s" %content)
            alert_obj.save()
```

**Comment by cgl** [comment]
FROM_EMAIL defined in settings, more comments inline

File: seshdash/models.py (hunk 4 above)
Comment (jlark) [inline seshdash/models.py]: Ok sorry
 it wasn't showing up.

Comment (cgl) [inline seshdash/models.py]: :) nope

**Pull request merged**